Break up account page, add activity page

wbazant opened the pull request:
Closes #767
Closes #766 
Closes #743 
Closes #695

ezwelty commented:
@wbazant This is great. Fun to see more big features coming together. The sections you've chosen to split the account into seem reasonable. So I'll focus my attention on the activity page.

1. For an account with as much activity as mine, the tag cloud is rather unwieldy. I could go on clicking for a long time, and the cloud becomes huge. A scrollable list would be more compact and quick to use.
2. The free-text search is neat. Does it work to hide instead of grey-out the changes shown within the same calendar day?
3. When the search results are few, the whole page jumps to the right.
4. I would rather the types matched the scheme we've been trying to make a standard elsewhere (common name + scientific name).
5. We should think about consistency in URLs and wording. Right now there is `/changes` and `/activity/{user.id}`, and `Recent changes` and `Activity` and `My activity`. Related aside: I think `/profiles/{user.id}` should be renamed to `/users/{user.id}` or `/people/{user.id}` since "profiles" is generic.
6. I also think that the single-user activity page should be a filtered global view. Literally something like `/changes?user_id={user.id}` (to which we could add other filters eventually, like `import_id` #424).

wbazant commented:
Thanks! We're going away today until May 8 but I'll try these when I'm back:

1. list types/places instead of the tag cloud: maybe make it similar to a types select menu? We would then search for a place / type, select it, and apply the filtering effect to the activity log. I'll see if I can still make the counts of most popular items easily visible, since they provide an overview of what the person is most interested in adding. 
2. hide instead of grey-out the changes shown within the same calendar day - will be better for a profile like yours, where changes in different locations aren't typically related
5. How about "Activity" being a header for /changes instead of "Recent changes", renaming /profiles/:userId to /people/:userId, and /activity/:userId to /people/:userId/activity? Mixing private and public paths under /users was very inconvenient, and the live site URLs are under there, so I renamed to /profiles, but /people seems nicer.


3. "When the search results are few, the whole page jumps to the right" I could not reproduce
4. Showing scientific names in 1., but leave activity log lines as they are, since it would take too much space. They're similar to map labels
6. Would be nice some day - right now I was only able to filter and facet since data for one user is small enough to all get fetched into the client, but we would need more infrastructure for providing that for the whole dataset. For #424 I think the really nice features are the map and querying by source, the activity log replaces https://fallingfruit.org/locations/home so it was fast to build with what we have now.

wbazant commented:
@ezwelty ready for review again! I've replaced the tags with two search boxes for type and place. Also, hiding unrelated entries from the same day, and I renamed header on recent changes page to 'Activity: recent changes' so it's similar to 'Activity: {userId}'.

I also fixed a bug with a back button where on desktop, /changes -> location -> another location -> [back] still went to the /changes page.

ezwelty commented:
@wbazant Thanks for making those changes. Looks good. I think "Activity" should be added to the user account dropdown (before or after "Account"?) and that the Global activity page header can just be "Activity" (which seems more consistent with "Activity: {user.name}" than "Activity: recent changes").

Could we use `/people/{user.id}` for public profiles and `/people/{user.id}/activity` for the activity pages? Since the public profiles (`/people/{user.id}`) are so empty, could they just contain the activity section rather than being an extra click away?
